### PR TITLE
Add support for specifying imagePullSecrets via resource configuration

### LINKED ISSUE #2467: Add support for specifying `imagePullSecrets` via resource configuration
### Component

JKube Kit

### Is your enhancement related to a problem? Please describe

This was asked by a user on [Gitter chat](https://matrix.to/#/!QryyIuhARjRcebsrTd:gitter.im/$iSrB9PfRllLNeF7W7LeIQtZX96VJKb59T7ui0tkqhZ0?via=gitter.im&via=matrix.org):

> How to configure the generated file to include the imagePullSecrets content in the maven pom file

As a user, If I want to specify [`imagePullSecrets`](https://kubernetes.io/docs/concepts/containers/images#specifying-imagepullsecrets-on-a-pod) for Deployment, I need to add them in resource fragment like this:

https://github.com/eclipse/jkube/blob/5dc19d9563adb3c10a2ad1f30e8f404d5ca29ab9/quickstarts/maven/tomee/src/main/jkube/deployment.yaml#L26-L27

This option should also be exposed via controller resource configuration

### Describe the solution you'd like

As a user, I  want to specify imagePullSecrets in my pom.xml like this:

```xml
<configuration>
  <resources> 
    <controller>
      <replicas>2</replicas>
       <imagePullSecrets>
           <imagePullSecret>${env.SECRET1}</imagePullSecret>
       </imagePullSecrets>
    </controller>
  </resources>
</configuration>
```
so that generated Deployment (or any other controller) contains specified imagePullSecrets
### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by arman-yekkehkhani**
## Description

Add support for specifying imagePullSecrets via resource configuration. Examples:
**Maven:**
```
<configuration>
  <resources> 
    <controller>
      <replicas>2</replicas>
       <imagePullSecrets>
           <imagePullSecret>${env.SECRET1}</imagePullSecret>
       </imagePullSecrets>
    </controller>
  </resources>
</configuration>
```
**Gradle**
```
resources {
    controller {
        imagePullSecrets = ['secret-1', 'secret-2']
    }
}
```
Fixes #2467 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [x] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3133 (_2024-06-05T06:38:51Z_)

:gear: [JKube E2E Tests (9379403368)](https://github.com/eclipse-jkube/ci/actions/runs/9379403368) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3133)](https://github.com/eclipse-jkube/ci/actions/runs/9379403368/job/25824282059)

File: gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ImagePullSecretsIT.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package org.eclipse.jkube.gradle.plugin.tests;
```
Comment (rohanKanojia) [inline gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ImagePullSecretsIT.java:1]: license header missing, could you please add it by running `mvn -N license:format`?

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
@arman-yekkehkhani : Thanks a lot, looks good to me.

Could you please add a line to CHANGELOG.md regarding this change?

File: gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ImagePullSecretsIT.java (hunk 1 above)
Comment (arman-yekkehkhani) [reply gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ImagePullSecretsIT.java:14]: thx! added

**Comment by arman-yekkehkhani** [comment]
@rohanKanojia Thanks Rohan. Added the license and updated CHANGELOG to reflect the new feature. Please take a look

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**